### PR TITLE
Add aws/auth action

### PR DESCRIPTION
**Pull request by reakaleek**
Was successfully tested in https://github.com/elastic/apm-aws-lambda/actions/runs/9597412100/workflow?pr=491

### CODE REVIEW

File: .github/workflows/test-aws-auth.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  contents: read
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./aws/auth
+        id: aws-auth
+        continue-on-error: true
```
Comment (v1v) [inline .github/workflows/test-aws-auth.yml:25]: why?

File: aws/auth/action.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+        repository = os.environ['REPOSITORY']
+        workflow_ref = os.environ['WORKFLOW_REF']
+        worflow_filename = workflow_ref.split('/')[4].split('@')[0]
+
+        m = hashlib.sha256()
+        m.update(f"{repository}/{worflow_filename}".encode('utf-8'))
+        hash = m.hexdigest()[:55]
+        role_name = f"gha-{hash}-role"
+        role_arn = f"arn:aws:iam::697149045717:role/{role_name}"
```
Comment (v1v) [inline aws/auth/action.yml:40]: Could you customise this? We may use some other AWS accounts in the near future

File: aws/auth/action.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        with open(os.environ['GITHUB_OUTPUT'], 'a') as f:
+            f.write(f"role-arn={role_arn}")
+
+    - name: Configure AWS Credentials for China region audience
```
Comment (v1v) [inline aws/auth/action.yml:44]: why?

File: aws/auth/action.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      uses: aws-actions/configure-aws-credentials@v4
```
Comment (v1v) [inline aws/auth/action.yml:45]: a pinned version?

File: aws/auth/action.yml (hunk 3 above)
Comment (reakaleek) [reply aws/auth/action.yml:44]: woops, copy paste

Comment (reakaleek) [reply aws/auth/action.yml]: https://github.com/elastic/oblt-actions/pull/63/commits/22eb7423f2a4f76368581c2b153e80c9a52d9267

File: aws/auth/action.yml (hunk 2 above)
Comment (reakaleek) [reply aws/auth/action.yml]: https://github.com/elastic/oblt-actions/pull/63/commits/0c3fed4536b58d0c2222f548f270329e90d03e8f

File: .github/workflows/test-aws-auth.yml (hunk 1 above)
Comment (reakaleek) [reply .github/workflows/test-aws-auth.yml:25]: I was only interested in testing the output.

But if you prefer, I can turn this into an integration test that actually tries to authenticate

File: aws/auth/action.yml (hunk 4 above)
Comment (reakaleek) [reply aws/auth/action.yml]: https://github.com/elastic/oblt-actions/pull/63/commits/0c3fed4536b58d0c2222f548f270329e90d03e8f#diff-3b9084e215925d2515c8a9bf5c1ef69b5d5b47a1e71ef91261b5f99fa622e904R51

File: .github/workflows/test-aws-auth.yml (hunk 1 above)
Comment (v1v) [reply .github/workflows/test-aws-auth.yml:25]: all good

**Review by v1v — APPROVED** [review]

**Pull request merged**